io/runes/runes_test.go

```diff
@@ -0,0 +1,177 @@
+package runes_test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+	"testing"
+
+	"github.com/madlambda/spells/assert"
+	"github.com/madlambda/spells/io/runes"
+	"github.com/madlambda/spells/iotest"
+)
+
+type testResult struct {
+	err     error
+	partial []rune
+}
+
+type testcase struct {
+	name   string
+	input  string
+	repeat int
+	want   testResult
+}
+
+const socraticParadox = "Εν οίδα ότι ουδέν οίδα"
+
+var testcases = []testcase{
+	{
+		name:   "empty input",
+		input:  "",
+		repeat: 1,
+	},
+	{
+		name:   "invalid code point",
+		input:  string([]byte{0x80}),
+		repeat: 1,
+		want: testResult{
+			err: fmt.Errorf("invalid rune at offset 0"),
+		},
+	},
+	{
+		name:   "simple ascii",
+		input:  "test",
+		repeat: 1,
+	},
+	{
+		name:   "single rune - 2 bytes",
+		input:  "ί",
+		repeat: 1,
+	},
+	{
+		name:   "1 ascii - start with multibyte",
+		input:  "ίA",
+		repeat: 1,
+	},
+	{
+		name:   "1 ascii - end with multibyte",
+		input:  "Aί",
+		repeat: 1,
```

Comment:
Maybe allow repeat to be default = 1 (undefined will be 0, then on the test case you set it to 1).